### DISCUSSION TITLE
xrepo env shell 钩子

### DISCUSSION THREAD

**Discussion by yipengsun** [xmake-io/xmake#4293] [Q&A]
请问可否在 xrepo env shell 加入钩子，用来执行任意用户命令，比如：

- 设置环境变量，如 `PYTHONENV`
- 载入 Python virtualenv

**Comment by waruqi** [comment]
> 请问可否在 xrepo env shell 加入钩子，用来执行任意用户命令，比如：
> 
> * 设置环境变量，如 `PYTHONENV`

```console
export PYTHONENV=xxx
xrepo env shell
```

> * 载入 Python virtualenv

```console
xrepo env -b "python 3.x" shell
```

or 

```console
cd xxx
edit xmake.lua -> add_requires("python 3.x")
xrepo env shell
```

**Reply by yipengsun** [reply]
目前我在 `xmake.lua` 里面已经写了 `add_requires("python")`, 但是不知道如何 **自动地** 设置环境变量。上面 `export PYTHONENV=xxx` 应该只能在 Linux 下用吧? 而且每次用 `xrepo env shell` 时都需要重新输入？

不知道能不能设置一个 target，然后用 `runenvs` 之类的东西创建类似的钩子（我没搜到 `runenvs` 的文档）？

**Reply by yipengsun** [reply]
~~换言之，就是想要一个类似于 [`direnv`](https://direnv.net/) 的功能。目前 direnv 不支持 windows。~~
`direnv` 现在貌似支持 windows 了，那我直接用 `direnv` 就好了。多谢。
